### PR TITLE
[BEAM-1456] Make UnboundedSourceWrapper snapshot to rescalable operator state in Flink Runner

### PR DESCRIPTION
**Pull request by JingsongLi**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by JingsongLi** [comment]
R: @aljoscha  Please take a look.

File: runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedSourceWrapper.java [hunk 1]
```diff
@@ -124,6 +130,9 @@
   private transient List<
```
Comment (aljoscha) [inline runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedSourceWrapper.java]: I think we don't need `restoredState` and the logic that does the copying in `initializeState()`. We should be able to just use `stateForCheckpoint` in `open()` for initialisation.

Comment (JingsongLi) [inline runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedSourceWrapper.java]: Yes, it is redundant.

**Comment by aljoscha** [comment]
Thanks for fixing this, @JingsongLi! I just merged 👍

**Pull request closed without merging**